lib/puppetserver/ca/action/list.rb

```diff
@@ -78,8 +87,17 @@ def run(input)
           all_certs = get_all_certs(puppet.settings).select { |cert| filter_names.call(cert) }
           requested, signed, revoked = separate_certs(all_certs)
           missing = certnames - all_certs.map { |cert| cert['name'] }
+          
+          if all && output_format == 'json'
+            grouped_cert = { "requested" => requested,
+                             "signed" => signed, 
+                             "revoked" => revoked,
+                             "missing" => missing }.to_json
+            puts grouped_cert
```

Comment:
This should use `@logger.inform` rather than `puts`. We try to route everything through our logger for better encapsulation.

Reply:
Gotcha. Thanks for letting me know!